File: linear_model_homework_1/program/R/3.5.R
dataframe = t(read.csv("C:/linear_model/data/3.5.txt",header = FALSE,sep = " "))
colnames(dataframe) = c("X","y")
row.names(dataframe) = c(1:nrow(dataframe))
dataframe = as.data.frame(dataframe)
head(dataframe,n = 3)

model=lm(y~X,data=dataframe)
summary(model)

qplot(dataframe$X, dataframe$y,size=20,color="r",xlab = expression(X),ylab = expression(y))

new.value = data.frame(15.3)
colnames(new.value) = "X"
predict(model, newdata = new.value)

File: linear_model_homework_1/program/python/3.5.py
# -*- coding: utf-8 -*-
"""
Created on Sun Nov 20 16:25:41 2016

@author: Windows98
"""

import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt


dataframe = np.loadtxt("C:/linear_model/data/3.5.txt").T
dataframe = sm.add_constant(dataframe)
dataframe[:3]


X = dataframe[:,:2]
y = dataframe[:,2]
regressor = sm.OLS(y, X).fit()
print(regressor.summary())

fig,ax = plt.subplots(figsize=(10,10))
ax.scatter(X[:,1], y , s=120, c="r")
ax.set_xlabel(u"$X$",fontsize=30)
ax.set_ylabel(u"$y$",fontsize=30)
ax.set_title(u"$y-X$",fontsize=30)

ploter = sns.JointGrid(x=X[:,1], y=y)
img = ploter.plot_joint(plt.scatter, s=60, color="g")
regressor.predict([1., 15.3])